auth-foundation/src/main/java/com/okta/authfoundation/client/internal/NetworkUtils.kt

```diff
@@ -127,36 +127,40 @@ private suspend fun OidcConfiguration.executeRequest(
 ): Response {
     currentCoroutineContext().ensureActive()
     val rateLimitErrorCode = 429
+    var event: RateLimitExceededEvent
```

Comment:
Can `event`, and `response` be local `val` inside the `do`?

Reply:
Response can't be local because I'm returning it outside the while loop. I tried returning it inside the while loop as well, but the compiler thinks it's not good enough to have it there.